src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/posthearingreview/PostHearingReviewAboutToSubmitHandler.java

```diff
@@ -37,15 +47,35 @@ public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType,
                                                           String userAuthorisation) {
         SscsCaseData caseData = callback.getCaseDetails().getCaseData();
 
-        PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(caseData);
-
         String caseId = caseData.getCcdCaseId();
 
         PostHearingReviewType typeSelected = caseData.getPostHearing().getReviewType();
         log.info("Review Post Hearing App: handling action {} for case {}", typeSelected,  caseId);
 
+        PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(caseData);
+
+        if (PostHearingReviewType.SET_ASIDE.equals(typeSelected)) {
+            addDocument(caseData, SET_ASIDE_APPLICATION);
+        }
+
         SscsUtil.clearDocumentTransientFields(caseData);
 
         return response;
     }
+
+    private void addDocument(SscsCaseData caseData, DocumentType documentType) {
+        DocumentLink url = caseData.getDocumentStaging().getPreviewDocument();
+        String now = LocalDate.now().format(DateTimeFormatter.ofPattern("dd-MM-yyyy"));
+        SscsDocumentTranslationStatus documentTranslationStatus = null;
+        if (caseData.isLanguagePreferenceWelsh()) {
```

Comment:
This is not suitable here as there could be multiple checks later on once other stories implemented. So at that time it can be changed to switch/case.

> `SscsDocumentTranslationStatus documentTranslationStatus = caseData.isLanguagePreferenceWelsh() ? SscsDocumentTranslationStatus.TRANSLATION_REQUIRED : null;`

Reply:
okie dokie!